src/ci/docker/run.sh

```diff
@@ -105,6 +105,23 @@ if [ -f "$docker_dir/$image/Dockerfile" ]; then
     # It seems that it cannot be the same as $IMAGE_TAG, otherwise it overwrites the cache
     CACHE_IMAGE_TAG=${REGISTRY}/${REGISTRY_USERNAME}/rust-ci-cache:${cksum}
 
+    # Docker build arguments.
+    build_args=(
+        "build"
+        "--rm"
+        "-t" "rust-ci"
+        "-f" "$dockerfile"
+        "$context"
+    )
+
+    # If the environment variable SCRIPT is defined,
+    # set the build argument SCRIPT_ARG to SCRIPT.
+    # In this way, we run the SCRIPT defined in CI,
+    # instead of the one defined in the Dockerfile.
+    if [ -n "${SCRIPT+x}" ]; then
+      build_args+=("--build-arg" "SCRIPT_ARG=${SCRIPT}")
+    fi
```

Comment:
I had to add `--build-arg` and I decided to also remove some duplicate code from the script, by extracting the `docker build` arguments in to the `build_args` variable